src/dmd/backend/elfobj.d

```diff
@@ -3802,12 +3803,14 @@ else
         off += 2;
         Offset(codseg) = off;
 
-        // put a reference into .ctors/.dtors each
+        // put a reference into .init_array/.fini_array each
         // needs to be writeable for PIC code, see Bugzilla 13117
         const int flags = SHF_ALLOC | SHF_WRITE | SHF_GROUP;
-        foreach (name; [".dtors.d_dso_dtor", ".ctors.d_dso_ctor"])
+        int type = SHT_FINI_ARRAY;
+        foreach (name; [".fini_array.d_dso_dtor", ".init_array.d_dso_ctor"])
         {
-            const cdseg = Obj_getsegment(name.ptr, null, SHT_PROGBITS, flags, _tysize[TYnptr]);
+            const cdseg = Obj_getsegment(name.ptr, null, type, flags, _tysize[TYnptr]);
+            type = SHT_INIT_ARRAY;
```

Comment:
This seems to be a slightly hacky way to write this.

maybe something like:

```d
type = name[1] == 'f' ? SHT_FINI_ARRAY : SHT_INIT_ARRAY;
```

would be better?

Reply:
This is irrelevant now. I just unrolled the loop to work around the bug which existed in dmd-2.079.0.